service/controller/resource/nodepool/create_deployment_uninitialized.go

```diff
@@ -136,11 +148,53 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 		return currentState, nil
 	default:
 		r.Logger.LogCtx(ctx, "level", "debug", "message", "template and parameters unchanged")
+
+		err := r.saveAzureIDsInCR(ctx, virtualMachineScaleSetsClient, virtualMachineScaleSetVMsClient, &azureMachinePool)
+		if err != nil {
+			r.Logger.LogCtx(ctx, "level", "debug", "message", "error trying to save object in k8s API", "stack", microerror.JSON(microerror.Mask(err)))
+			r.Logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return currentState, nil
+		}
+
 		r.Logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 		return currentState, nil
 	}
 }
 
+func (r *Resource) saveAzureIDsInCR(ctx context.Context, virtualMachineScaleSetsClient *compute.VirtualMachineScaleSetsClient, virtualMachineScaleSetVMsClient *compute.VirtualMachineScaleSetVMsClient, azureMachinePool *capzexpv1alpha3.AzureMachinePool) error {
+	r.Logger.LogCtx(ctx, "level", "debug", "message", "saving provider status info in CR")
+
+	vmss, err := virtualMachineScaleSetsClient.Get(ctx, key.ClusterID(azureMachinePool), key.NodePoolVMSSName(azureMachinePool))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	instances, err := r.GetVMSSInstances(ctx, virtualMachineScaleSetVMsClient, key.ClusterID(azureMachinePool), key.NodePoolVMSSName(azureMachinePool))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	provisioningState := capzv1alpha3.VMState(*vmss.ProvisioningState)
+	azureMachinePool.Status.ProvisioningState = &provisioningState
+	azureMachinePool.Status.Replicas = int32(len(instances))
+	azureMachinePool.Spec.ProviderID = fmt.Sprintf("azure://%s", *vmss.ID)
+
+	providerIDList := make([]string, len(instances))
+	for i, vm := range instances {
+		providerIDList[i] = fmt.Sprintf("azure://%s", *vm.ID)
+	}
+	azureMachinePool.Spec.ProviderIDList = providerIDList
+
+	err = r.CtrlClient.Update(ctx, azureMachinePool)
```

Comment:
Does calling `Update` store the `Status` fields as well? I thought `UpdateStatus` was required in this case.

Reply:
You are right. I'll fix it.